src/main/scala/com.snowplowanalytics.sauna/apis/UrbanAirship.scala

```diff
@@ -0,0 +1,166 @@
+package com.snowplowanalytics.sauna
+package apis
+
+
+
+import java.io.DataOutputStream
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext
+import scala.util.{ Failure, Success }
+import org.apache.commons.io.IOUtils
+import java.io.File
+import java.util.Properties
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.io.InputStream
+import java.util.ArrayList
+import java.io.InputStream
+import java.util.HashMap
+import java.util.Iterator
+import scala.io.Source.fromFile
+import java.net.URL
+import akka.actor.{ActorRef, Props}
+import java.util.Map.Entry
+import scala.io.Source
+import scala.collection
+import scala.collection.mutable.{ Map => MutableMap }
+import javax.net.ssl.HttpsURLConnection
+import java.util.zip.GZIPOutputStream
+import org.json.JSONObject
+import responders.urbanairship.UAResponder.Airship
+import scala.collection.immutable.ListMap
+import play.api.libs.json.Json
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class UrbanAirship(implicit logger: ActorRef) {
```

Comment:
It's good habit to always annotate your classes and methods with docstrings. Even if you think class/method purpose is obvious.
When you formulating function in natural language it helps you to understand more about it, spot things that are not obvious (like side-effects) and cherry-pick its essence.